R/sleuth.R

```diff
@@ -284,16 +283,66 @@ sleuth_prep <- function(
     ret$obs_norm_filt <- dplyr::semi_join(obs_norm, filter_df, by = 'target_id')
     ret$tpm_sf <- tpm_sf
 
+    path <- kal_dirs[1]
+    kal_path <- get_kallisto_path(path)
+    target_id <- as.character(rhdf5::h5read(kal_path$path, "aux/ids"))
+    num_transcripts <- length(target_id)
+    ret$bs_quants <- list()
+
+    msg("calculating bootstrap quantiles")
+    ret$bs_quants <- lapply(1:length(kal_dirs), function(i) {
+      kal_path <- get_kallisto_path(kal_dirs[i])
+      num_bootstrap <- as.integer(rhdf5::h5read(kal_path$path, "aux/num_bootstrap"))
+      msg(paste0("reading sample: ",names(kal_dirs[i])))
+      read_bootstrap_quant(fname = kal_path$path, num_bootstrap)
+    })
+    names(ret$bs_quants) <- names(kal_dirs)
+
+    transformation_function <- function(x) log(x + 0.5)
+
+    msg("calculating bootstrap variance")
+    ret$bs_summary <- do.call(cbind, lapply(seq_along(ret$kal), function(i) {
+      path <- kal_dirs[i]
+      samp_name <- names(path)
+      kal_path <- get_kallisto_path(path)
+      num_bootstrap <- as.integer(rhdf5::h5read(kal_path$path, "aux/num_bootstrap"))
+      bs_var <- read_bootstrap_statistics(
+        fname = kal_path$path,
+        num_bootstrap = num_bootstrap,
+        est_count_sf = est_counts_sf[[i]],
+        num_transcripts = num_transcripts,
+        transform = transformation_function
+        )
+      dot(i)
+      bs_var
+    }))
+
+    ret$target_id <- target_id
+
+    bs_test_summary <- adf(ret$bs_summary)
+    bs_test_summary$target_id <- target_id
+    bs_test_summary <- bs_test_summary[order(bs_test_summary$target_id), ]
+    bs_test_summary <- data.frame(varMeans =
+        rowMeans(bs_test_summary[,names(bs_test_summary) != "target_id"]), target_id =
+        bs_test_summary$target_id)
+
     msg('summarizing bootstraps')
-    # TODO: store summary in 'obj' and check if it exists so don't have to redo every time
-    bs_summary <- bs_sigma_summary(ret, function(x) log(x + 0.5))
-    # TODO: check if normalized. if not, normalize
-    # TODO: in normalization step, take out all things that don't pass filter so
-    # don't have to filter out here
-    bs_summary$obs_counts <- bs_summary$obs_counts[ret$filter_df$target_id, ]
-    bs_summary$sigma_q_sq <- bs_summary$sigma_q_sq[ret$filter_df$target_id]
-
-    ret$bs_summary <- bs_summary
+    bs_test_summary <- list(sigma_q_sq = bs_test_summary)
+
+    path <- kal_dirs[1]
+    kal_path <- get_kallisto_path(path)
+    obs_counts <- obs_to_matrix(ret, "est_counts")
+    obs_counts <- transformation_function(obs_counts)
+
+    bs_test_summary$obs_counts <- obs_counts[rownames(obs_counts)
```

Comment:
is it just me, or are these filter commands a bit obscure? are they not simply the same filter command?